quixstreams/state/memory/partition.py

```diff
@@ -0,0 +1,142 @@
+import logging
+
+from typing import Optional, Dict, Any, Union
+
+from quixstreams.models import ConfluentKafkaMessageProto
+from quixstreams.state.recovery import ChangelogProducer
+from quixstreams.state.base import StorePartition, PartitionTransactionCache
+from quixstreams.state.metadata import METADATA_CF_NAME
+from quixstreams.state.exceptions import ColumnFamilyDoesNotExist
+from quixstreams.utils.json import loads as json_loads, dumps as json_dumps
+
+logger = logging.getLogger(__name__)
+
+__all__ = ("MemoryStorePartition",)
+
+
+class MemoryStorePartition(StorePartition):
+    """
+    Class to access in-memory state.
+
+    Responsibilities:
+     1. Recovering from changelog messages
+     2. Creating transaction to interact with data
+     3. Track partition state in-memory
+    """
+
+    def __init__(self, changelog_producer: Optional[ChangelogProducer]) -> None:
+        super().__init__(
+            dumps=json_dumps,
+            loads=json_loads,
+            changelog_producer=changelog_producer,
+        )
+        self._processed_offset: Optional[int] = None
+        self._changelog_offset: Optional[int] = None
+        self._state: Dict[str, Dict[bytes, Dict[bytes, Any]]] = {
+            "default": {},
+            METADATA_CF_NAME: {},
+        }
```

Comment:
The type hint seems to be wrong here: 
Looks like it should be `Dict[str, Dict[bytes, Any]]` instead